include/deal.II/lac/constrained_linear_operator.h

```diff
@@ -0,0 +1,292 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__constrained_linear_operator_h
+#define dealii__constrained_linear_operator_h
+
+#include <deal.II/lac/linear_operator.h>
+#include <deal.II/lac/packaged_operation.h>
+#include <deal.II/lac/constraint_matrix.h>
+
+#ifdef DEAL_II_WITH_CXX11
+
+DEAL_II_NAMESPACE_OPEN
+
+
+/**
+ * @name Indirectly applying constraints to LinearOperator
+ */
+//@{
+
+
+/**
+ * This function takes a ConstraintMatrix @p constraint_matrix and an
+ * operator exemplar @p exemplar (this exemplar is usually a linear
+ * operator that describes the system matrix) and returns a LinearOperator
+ * object associated with the "homogeneous action" of the underlying
+ * ConstraintMatrix object:
```

Comment:
Oh, scratch that -- I misread that this isn't actually the function that represents the constrained operator.

In that case, can you explain in the documentation why one needs to provide such an exemplar to begin with, and stress that the *action* of the exemplar is never used anywhere?